include/extensions.h

```diff
@@ -9,13 +9,18 @@
 struct background_config {
         wlc_handle output;
         wlc_resource surface;
-        struct wl_resource *resource;
+        // we need the wl_resource of the surface in the destructor
+        struct wl_resource *wl_surface_res;
 };
 
 struct panel_config {
+        // wayland resource used in callbacks, is used to track this panel
+        struct wl_resource *wl_resource;
         wlc_handle output;
         wlc_resource surface;
-        struct wl_resource *resource;
+        // we need the wl_resource of the surface in the destructor
```

Comment:
These comments are the sort of thing that would be better to put into the extended commit message.